daemon/volumes_unix.go

```diff
@@ -316,7 +316,7 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 			}
 
 			if len(cp.Source) == 0 {
-				v, err := daemon.createVolume(cp.Name, cp.Driver, nil)
+				v, err := daemon.volumes.GetWithRef(cp.Name, cp.Driver, container.ID)
```

Comment:
In some cases we are creating (or registering a container after daemon restart) a new volume with a container and its racey to create it, then increment, and I liked it better than passing an empty string in "Create" and "Get"